compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -212,6 +212,10 @@ struct ValidityVisitor<'rt, 'mir, 'tcx, M: Machine<'mir, 'tcx>> {
 }
 
 impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, 'tcx, M> {
+    fn allow_uninit_and_ptr_numbers(&self) -> bool {
+        self.ctfe_mode.is_none() && !M::enforce_number_initialization(self.ecx)
```

Comment:
I wonder if it would make sense to not use ctfe_mode here, and instead set `enforce_number_initialization` appropriately for the CTFE `Machine` instance? That seems less redundant.

Reply:
Sure, I can try that. What code should I change?

Reply:
Basically, change this function to `!M::enforce_number_initialization(self.ecx)` (and then probably we can just inline it whereever it is needed). And then in the `compile_time_machine` macro, make `enforce_number_initialization` return `true`.